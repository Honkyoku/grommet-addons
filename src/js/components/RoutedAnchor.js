// (C) Copyright 2014-2015 Hewlett Packard Enterprise Development LP

import React, { Component, PropTypes } from 'react';
import classnames from 'classnames';
import Anchor from 'grommet/components/Anchor';

/**
* This component is used to augment Grommet anchor
* with routing/history capabilities.
*/
export default class RoutedAnchor extends Component {

  constructor () {
    super();
    this._onClick = this._onClick.bind(this);
  }

  _onClick (event) {
    const { method, onClick, path} = this.props;
    const { router } = this.context;

    event.preventDefault();

    if ('push' === method) {
      router.push(path);
    } else if ('replace' === method) {
      router.replace(path);
    }

    if (onClick) {
      onClick();
    }
  }

  render () {
    const { className, path, ...props } = this.props;
    const { router } = this.context;
    let classes = classnames(
      { [`active`]: router.isActive(path) },
      className
    );
    let href = router.createPath(path);
    return (
      <Anchor {...props} className={classes} href={href}
        onClick={this._onClick} />
    );
  }
};

RoutedAnchor.propTypes = {
  method: PropTypes.oneOf(['push', 'replace']),
  path: PropTypes.string.isRequired
};

RoutedAnchor.defaultProps = {
  method: 'push'
};

RoutedAnchor.contextTypes = {
  router: PropTypes.object.isRequired
};
