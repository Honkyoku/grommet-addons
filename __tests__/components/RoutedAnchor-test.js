// (C) Copyright 2014-2016 Hewlett Packard Enterprise Development LP

import React from 'react';
import renderer from 'react/lib/ReactTestRenderer';

import RoutedAnchor from '../../src/js/components/RoutedAnchor';
import RouterContext from '../utils/RouterContext';

describe('RoutedAnchor', () => {
  it('has correct default options', () => {
    const component = renderer.create(
      <RouterContext>
        <RoutedAnchor path='test' />
      </RouterContext>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
