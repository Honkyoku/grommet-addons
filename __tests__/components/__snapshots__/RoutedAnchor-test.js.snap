exports[`RoutedAnchor has correct default options 1`] = `
<div>
  <a
    aria-label={undefined}
    className="grommetux-anchor"
    href={undefined}
    method="push"
    onClick={[Function bound _onClick]} />
</div>
`;
